azext_iot/iothub/providers/base.py

```diff
@@ -29,3 +33,43 @@ def __init__(self, cmd, hub_name, rg, login=None, auth_type_dataplane=None):
 
     def get_sdk(self, sdk_type):
         return self.resolver.get_sdk(sdk_type)
+
+
+class IoTHubResourceProvider(object):
+    """Class to use for control plane operations."""
+    def __init__(
+        self,
+        cmd,
+        hub_name: str,
+        rg: Optional[str] = None,
+    ):
+        self.cmd = cmd
+        self.api_version = "2022-04-30-preview"
```

Comment:
This is more of a conversation starter than a change request - but I'm curious how this affects our attachment to specific versions of the CLI core / hub SDK.

Assuming we're using the management SDK from CLI core, a few notes:
- We likely won't have SDKs generated for preview API versions, or if they are - they will have to be specified manually (like done here) and there may be some issues with importing models for these calls as they may be different from the non-preview Hub models
- We also need to keep track of which API-versions are in which versions of core, and we'll need to wait until CLI core is released with this new API-version before we ship the extension.
- It will also affect our min-CLI versions, as we're only as backwards-compatible as the API-versions available in whichever version of the management SDK that version of CLI core uses
- We *could* try to use the latest default version from CLI core, but that seems like a recipe for being broken by core updates without us knowing :)

Reply:
tbh it would be good if we can rely on the most current api version from the core. Otherwise, if no sdk is provided, it pulls from the api list in core and that is not the preview one that we want. The sdk is whatever is installed (I added a temporary azure-mgmt-iothub==2.3.0 to the setup.py)

I was going to make a pr in core to update the min api + force the azure-mgmt-iothub to be 2.3.0 and then in the extension's setup.py force the min cli core to be whatever cli core version that has the change in (and remove our dependency on azure-mgmt-iothub and the hardcoded api version since core will handle both)

There may be a few hiccups since you could technically install and uninstall packages and somehow get into a weird state where you have this extension with an old package. Would it make sense to have a check for "do you have the min package installed?" like how we do for track 1 vs 2?

as for your notes:
1. yes this will be an issue but as long as we say "for this feature and beyond we want min cli core version x that has sdk version y" we should be fine.
2. yes. It will be annoying but it will be done
3. yes, the new min cli version will be something very new so it may annoy some customers. But this is what happens when an api is not backwards compatible.
4. are you saying we should keep this hard coded api version? Id rather not because I would like to not have to track these here... Also will there be a cli core update touching iot (in a meaningful way) without us knowing?

Reply:
Ok, we can throw the dependency on core out the window since the api is now backwards compatible. From my testing, I have not found anything of issue (aside from being able to create and test routes with the cosmos db endpoint and then not list/update/show/delete them)

So it seems like all 4 points don't apply anymore...